File: go/make.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"path/filepath"
)

func main() {
	args := os.Args
	if len(args) < 2 {
		fmt.Println("ERROR! No day argument given. Usage: make [day]")
		os.Exit(1)
	}
	day := args[1]

	wd, err := os.Getwd()
	if err != nil {
		panic(err)
	}
	path := filepath.Join(wd, day)

	err = os.Mkdir(path, os.ModePerm)
	if err != nil {
		panic(err)
	}

	input_dir := filepath.Join(wd, "../input/", day)
	err = os.Mkdir(input_dir, os.ModePerm)
	if err != nil {
		panic(err)
	}

	os.Create(filepath.Join(input_dir, "example.txt"))
	os.Create(filepath.Join(input_dir, "question.txt"))
	file, err := os.Create(fmt.Sprintf("%s/%s", path, "part1.go"))
	if err != nil {
		panic(err)
	}

	writer := bufio.NewWriter(file)
	writer.WriteString(fmt.Sprintf("package main\n\nfunc main() {\npath, _ := filepath.Abs(\"../input/%s/question.txt\")\nfile, err := os.Open(path)\nif err != nil {\npanic(err)\n}\ndefer file.Close()\n\nscanner := bufio.NewScanner(file)\n}\n", day))
	writer.Flush()
}

File: zig/make.zig
const std = @import("std");

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    var arena = std.heap.ArenaAllocator.init(gpa.allocator());
    defer arena.deinit();
    const allocator = arena.allocator();

    const args = try std.process.argsAlloc(allocator);
    if (args.len < 2) {
        std.log.err("No day argument given. Usage: zig run <file.zig> -- [day]", .{});
        std.process.abort();
    }
    const day = args[1];

    const cwd = try std.process.getCwdAlloc(allocator);

    const path = try std.fs.path.join(allocator, &[_][]const u8{cwd, day});
    std.os.mkdir(path, 511) catch |err| {
        if (err != std.os.MakeDirError.PathAlreadyExists) {
            return err;
        }
        std.log.info("folder '{s}' already exists, skipping creation", .{path});
    };

    const input_dir = try std.fs.path.join(allocator, &[_][]const u8{cwd, "../input/", day});
    std.os.mkdir(input_dir, 511) catch |err| {
        if (err != std.os.MakeDirError.PathAlreadyExists) {
            return err;
        }
        std.log.info("folder '{s}' already exists, skipping creation", .{input_dir});
    };

    const example_path = try std.fs.path.join(allocator, &[_][]const u8{input_dir, "example.txt"});
    _ = try std.fs.createFileAbsolute(example_path, .{.read = true});

    const question_path = try std.fs.path.join(allocator, &[_][]const u8{input_dir, "question.txt"});
    _ = try std.fs.createFileAbsolute(question_path, .{.read = true});

    const code_path = try std.fs.path.join(allocator, &[_][]const u8{path, "part1.zig"});
    _ = try std.fs.createFileAbsolute(code_path, .{.read = true});
}